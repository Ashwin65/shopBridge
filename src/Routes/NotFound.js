import React, { Component } from 'react';
import { Button, Result } from 'antd';
import 'antd/dist/antd.css';
import history from './history';


export class PageNotFound extends Component {
  /**
   * @description redirects the user back to home page based on the role
   */
  handleClick = () => {
        console.log("handle click")
          history.back();
        }
    


  render() {
    return (
      <>
        <Result
          status={'404'}
          title={'404'}
          subTitle={'Sorry, the page you visited does not exist.'}
          extra={
            <Button
              id="home_button"
              type="primary"
              key="console"
              onClick={this.handleClick}
            >
              Back Home
            </Button>
          }
        />
      </>
    );
  }
}


export default PageNotFound;
