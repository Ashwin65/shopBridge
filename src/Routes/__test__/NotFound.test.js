import { shallow, mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import history from '../history';
import { PageNotFound } from '../NotFound';
const pushSpy = jest.spyOn(history, 'back');

describe('Test for NotFound', () => {
  it('Snapshot should Match', () => {
    const container = mount(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});



describe('Testing functions of not found', () => {
  const container = shallow(<PageNotFound  />);
  const testInstance = container.instance();
  it('should test the handleclick function when user is a super admin', () => {
    testInstance.handleClick();
    expect(pushSpy).toHaveBeenCalled();
  });
});
