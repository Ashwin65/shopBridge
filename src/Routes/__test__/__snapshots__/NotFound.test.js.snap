// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NotFound Snapshot should Match 1`] = `ReactWrapper {}`;
